### PR TITLE
Issue #1723: Pedigree drag and drop UI needs better feedback

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

File: components/pedigree/resources/src/main/resources/pedigree/view/legend.js [hunk 1]
```diff
@@ -308,6 +309,34 @@ var Legend = Class.create( {
...
+     */
+    _onHoverWrapper: function(label, target, overlap, event) {
+        if (editor.isReadOnlyMode()) {
+            return;
+        }				
+        var divPos = editor.getWorkspace().viewportToDiv(event.pointerX(), event.pointerY());
+        var pos    = editor.getWorkspace().divToCanvas(divPos.x,divPos.y);
+        var node   = editor.getView().getPersonNodeNear(pos.x, pos.y);
+        if (node) {
+			node.getGraphics().getHoverBox().setHighlighted(true);
```
Comment (sdumitriu) [inline components/pedigree/resources/src/main/resources/pedigree/view/legend.js]: Don't use tabs for indentation.

File: components/base-war/src/main/webapp/resources/js/dragdrop.js [hunk 2]
```diff
@@ -0,0 +1,974 @@
+// script.aculo.us dragdrop.js v1.9.0, Thu Dec 23 16:54:48 -0500 2010
```
Comment (sdumitriu) [inline components/base-war/src/main/webapp/resources/js/dragdrop.js]: This should be in a `scriptaculous` subfolder: `components/base-war/src/main/webapp/resources/js/scriptaculous/dragdrop.js`

Comment (sdumitriu) [inline components/base-war/src/main/webapp/resources/js/dragdrop.js]: Also, make a pull request against [this file](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/scriptaculous/dragdrop.js).

Comment (veronikaslc) [inline components/base-war/src/main/webapp/resources/js/dragdrop.js]: Have "Permission to xwiki/xwiki-platform.git denied to veronikaslc"

Comment (sdumitriu) [inline components/base-war/src/main/webapp/resources/js/dragdrop.js]: Yes, you need to make a fork first.

**Comment by veronikaslc** [comment]
@allasm @DeanWay,   please review, the change is quite simple - the _onHoverWrapper() added to onHover() callback to highlight the node.

File: components/pedigree/resources/src/main/resources/pedigree/view/legend.js [hunk 3]
```diff
@@ -55,7 +56,7 @@ var Legend = Class.create( {
         });
 
         if (allowDrop) {
-            Droppables.add(editor.getWorkspace().canvas, {accept: 'drop-'+this._getPrefix(), onDrop: this._onDropWrapper.bind(this)});
+            Droppables.add(editor.getWorkspace().canvas, {accept: 'drop-'+this._getPrefix(), onDrop: this._onDropWrapper.bind(this), onHover: this._onHoverWrapper.bind(this)});
```
Comment (DeanWay) [inline components/pedigree/resources/src/main/resources/pedigree/view/legend.js]: perhaps you should separate this into multiple lines for readability

**Comment by DeanWay** [comment]
Just built this branch to manually test the drag-drop feature. Dragging disorders, genes, phenotypes and cancers from the sidebar no longer works as intended. The selected disorder, gene, etc. does is not displayed as dragging with the cursor

**Pull request closed without merging**